.github/workflows/publish.yml

```
@@ -11,23 +11,34 @@ on:
     paths:
       - "src/**"
       - ".github/workflows/publish.yml"
-  repository_dispatch:
-    types: [create-release]
-    client_payload:
+  workflow_dispatch:
+    inputs:
       release_tag:
         required: true
+        description: 'Tag for the new release'
+    dry_run:
+      description: 'If true, the draft release will not be finalized'
+      type: boolean
+      required: true
+      default: false
+
 
 jobs:
   create_release:
-    if: github.event_name == 'repository_dispatch'
+    if: github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-latest
+    outputs:
+      release_id: ${{ steps.create.outputs.release_id }}
     steps:
       - name: Create release
         env:
           RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
           GH_TOKEN: ${{ github.token }}
+        shell: pwsh
         run: |
-          gh release create "$RELEASE_TAG" --draft --generate-notes --title "Release $RELEASE_TAG"
+          $releaseUrl = gh release create "$RELEASE_TAG" --draft --generate-notes --title "Release $RELEASE_TAG"
```

this is PowerShell
```suggestion
          $releaseUrl = gh release create "$env:RELEASE_TAG" --draft --generate-notes --title "Release $RELEASE_TAG"
```

also could be nice to check the exit code
```suggestion
          $releaseUrl = gh release create "$env:RELEASE_TAG" --draft --generate-notes --title "Release $RELEASE_TAG"
          if ( $LASTEXITCODE -ne 0 ) { throw "Failed to create draft release" }
```
PowerShell's failfast mode doesn't automatically check exist code from external executables (unlike bash).

---

Oops sorry about those. I think everything has the right syntax now